include/flatbuffers/array.h

```diff
@@ -238,6 +238,12 @@ const Array<E, length> &CastToArrayOfEnum(const T (&arr)[length]) {
   return *reinterpret_cast<const Array<E, length> *>(arr);
 }
 
+template<typename T, uint16_t length>
+bool operator==(const Array<T, length> &lhs, const Array<T, length> &rhs) {
```

Comment:
How about passing in by pointer instead? You will have to nullptr check below, but you can then skip the memcp if the pointers are the same. It will also make the generated code the same, instead of adding the dereferencing part.

Reply:
Actually, it is not possible to be null, so unnecessary nullity check below.
What about keep operator== definitions same ? :)

Reply:
I don't get your second sentence.

Reply:
I mean, keep operator== the way accept reference instead of pointer

Reply:
The downside is if you compare two arrays to themselves, you do additional work comparing the elements.

How about leave the reference parameter and just add an explicit check for their address to see if they are the same object?